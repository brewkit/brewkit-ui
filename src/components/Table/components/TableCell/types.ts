
// import { TableCellProps } from "@material-ui/core/TableCell";


export type Props = {
    className?: string,
    children?: any,
    sortData?: any,
};
