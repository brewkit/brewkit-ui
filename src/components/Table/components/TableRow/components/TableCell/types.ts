import { HTMLProps, ReactNode } from 'react';


export interface Props extends Omit<HTMLProps<HTMLTableCellElement>, 'label'> {


    /**
     * The custom HTML element to use for the TableCell
     * 
     * @default td
     */
    as?: 'td' | 'th',


    /**
     * The contents of the TableCell
     */
    children?: ReactNode,


    /**
     * The className for the TableCell
     */
    className?: string,


    /**
     * The label to show in the TableCell
     */
    label?: ReactNode,


}
