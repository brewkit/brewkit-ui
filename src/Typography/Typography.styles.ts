import { createStyles, makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(
    (theme) =>
        createStyles({
            primary: { color: theme.palette.primary.main },
            secondary: { color: theme.palette.secondary.main },
            success: { color: theme.palette.success.main },
            error: { color: theme.palette.error.main },
            warning: { color: theme.palette.warning.main },
            text: { color: theme.palette.text.primary },
        }),
    { name: 'Typography' },
);

export default useStyles;
