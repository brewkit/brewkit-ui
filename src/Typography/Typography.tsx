import React from 'react';
import clsx from 'clsx';
import { merge } from 'lodash';
import {
    Typography as MUITypography,
    TypographyProps as MUITypographyProps,
} from '@material-ui/core';
import withoutKeys from '../utils/withoutKeys';
import useStyles from './Typography.styles';

export interface BkTypographyProps extends Omit<MUITypographyProps, 'color'> {
    color?:
        | 'primary'
        | 'secondary'
        | 'error'
        | 'success'
        | 'warning'
        | 'textPrimary'
        | 'textSecondary';
}

const Typography = React.forwardRef(
    (props: BkTypographyProps, ref: React.Ref<HTMLElement>) => {
        const {
            children,
            className,
            classes: userClasses = {},
            color = 'textPrimary',
            variant = 'body1',
            ...otherProps
        } = props;
        const typographyClasses = useStyles();
        const classes = merge(typographyClasses, userClasses);
        const muiClasses = withoutKeys(classes, [
            'error',
            'primary',
            'secondary',
            'success',
            'textPrimary',
            'textSecondary',
            'warning',
        ]);

        return (
            <MUITypography
                className={clsx(className, classes?.[color])}
                classes={muiClasses}
                color={undefined} // We manually resolve colors color
                ref={ref}
                variant={variant}
                {...otherProps}>
                {children}
            </MUITypography>
        );
    },
);

Typography.displayName = 'BkTypography';

export default Typography;
