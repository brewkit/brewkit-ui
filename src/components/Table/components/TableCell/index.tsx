
import * as React from 'react';
import clsx from 'clsx';
import TableCell from '@material-ui/core/TableCell';
import TableSortLabel from '@material-ui/core/TableSortLabel';
import styles from './styles';
import { Props } from './types';


export const TableCellComponent = React.forwardRef(({
    children,
    className,
    sortData,
    ...otherProps
}: Props, ref: React.Ref<any>): React.ReactElement => {


    const classes = styles();
    const classnames = clsx(
        classes.brewTableCell,
        className,
    );


    return (
        <TableCell
            className={classnames}
            {...otherProps}
            ref={ref}
        >
            {sortData &&
                <TableSortLabel active={sortData.active} direction={sortData.direction}>
                    {sortData.label}
                </TableSortLabel>
            }
            {children}
        </TableCell>
    );

});


export default TableCellComponent;
