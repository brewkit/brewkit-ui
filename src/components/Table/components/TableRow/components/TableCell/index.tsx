import React, { HTMLProps, ReactElement } from 'react';
import clsx from 'clsx';
import { Props } from './types';


/**
 * The `<TableCell />` component represents a single cell in a `<TableRow />`
 */
function TableCell({
    as,
    className,
    children,
    label,
    ...otherProps
}: Props): ReactElement {


    const classes = clsx(
        'brew-TableCell',
        className,
    );


    const CellElement: 'td' | 'th' = as || 'td';


    const content = children || label || '';


    return (
        <CellElement className={classes} {...otherProps}>
            {content}
        </CellElement>
    );


}


export default TableCell;
