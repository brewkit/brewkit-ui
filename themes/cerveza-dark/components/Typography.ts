

import { color } from '../../cerveza/variables';


const Typography = {


    overrides: {

        MuiTypography: {

            subtitle1: {
                color: color.blue.light,
            },

            subtitle2: {
                color: color.blue.light,
            },

            button: {
                color: color.blue.main,
            },

            body1: {
                color: color.blueGray.light,
            },

            body2: {
                color: color.blueGray.light,
            },

            caption : {
                color: color.blueGray.light,
            }

        },

    },


};


export default Typography;
