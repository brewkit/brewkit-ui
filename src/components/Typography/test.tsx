import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Typography from './index';


it('renders correctly', () => {
    const tree = renderer
        .create(<Typography>Foo</Typography>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


it('renders children when passed in', () => {
    const wrapper = shallow((
        <Typography>
            <span>foo</span>
        </Typography>
    ));
    expect(wrapper.contains(<span>foo</span>)).toEqual(true);
});


it('accepts custom classes', () => {
    const wrapper = shallow((
        <Typography className="customClass">Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('customClass')).toEqual(true);
    expect(wrapper.hasClass('brew-Typography')).toEqual(true);
});


it('"variant" defaults to "inherit" and applies appropriate class', () => {
    const wrapper = shallow((
        <Typography>Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('brew-Typography--variant-inherit')).toEqual(true);
});


it('"color" defaults to "initial" and applies appropriate class', () => {
    const wrapper = shallow((
        <Typography>Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('brew-Typography--color-initial')).toEqual(true);
});


it('"display" defaults to "initial" and applies appropriate class', () => {
    const wrapper = shallow((
        <Typography>Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('brew-Typography--display-initial')).toEqual(true);
});


it('"align" defaults to "inherit" and applies appropriate class', () => {
    const wrapper = shallow((
        <Typography>Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('brew-Typography--align-inherit')).toEqual(true);
});
