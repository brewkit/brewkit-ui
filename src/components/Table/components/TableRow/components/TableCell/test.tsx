import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TableCell from './index';
import Typography from '../../../../../Typography';


describe('Table Cell', () => {


    it('Renders correctly with text children', () => {
        const content = renderer
            .create(<TableCell>Column1</TableCell>)
            .toJSON();

        expect(content).toMatchSnapshot();
    });


    it('Renders correctly with component children', () => {
        const content = renderer
            .create(<TableCell><Typography variant="h3">Label</Typography></TableCell>)
            .toJSON();

        expect(content).toMatchSnapshot();
    });


    it('Includes custom className', () => {
        const className = `class-${String(Math.floor(Math.random() * 1000))}`;
        const content = shallow(<TableCell className={className}>Column1</TableCell>);

        expect(content.hasClass(className)).toBe(true);
        expect(content.hasClass('brew-TableCell')).toBe(true);
    });


});

describe('Table Cell Props', () => {


    it('Renders `as` prop correctly', () => {
        const content = renderer
            .create(<TableCell as="th">Column1</TableCell>)
            .toJSON();

        expect(content).toMatchSnapshot();

    });


});
