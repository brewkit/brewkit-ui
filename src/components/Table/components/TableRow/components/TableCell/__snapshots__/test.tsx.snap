// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Cell Props Renders \`as\` prop correctly 1`] = `
<th
  className="brew-TableCell"
>
  Column1
</th>
`;

exports[`Table Cell Renders correctly with component children 1`] = `
<td
  className="brew-TableCell"
>
  <h3
    className="brew-Typography brew-Typography--variant-h3 brew-Typography--color-initial brew-Typography--align-inherit brew-Typography--display-initial"
  >
    Label
  </h3>
</td>
`;

exports[`Table Cell Renders correctly with text children 1`] = `
<td
  className="brew-TableCell"
>
  Column1
</td>
`;
